### ISSUE TITLE
Deprecation warning from Jedi

### ISSUE THREAD

**Issue by charlesdong1991** [ipython/ipython#12102]
xref https://github.com/pandas-dev/pandas/pull/31323 and https://github.com/pandas-dev/pandas/issues/31342

It looks like there is a deprecation warning in Ipython once starting to use the latest jedi version (0.16.0)

**Comment by Carreau** [comment]
Thanks, i'll try to make a fix when I have time. 

Help welcome.

**Comment by JBKahn** [comment]
If it helps, this is the line I think the issue is wwith but I don't know the API on the other side, or see a specific unit test to hack on to ensure I don't break anything.

The line: https://github.com/ipython/ipython/blob/00f0d8afe7a585a78092ddf14d62fabd113da8e3/IPython/core/completer.py#L989

**Comment by aiguofer** [comment]
a quick bit of playing around with jedi, I think changing the code above to the following should fix it:

```
    return '(%s)'% ', '.join([f for f in (_formatparamchildren(p) for signature in completion.get_signatures() for p in signature.defined_names()) if f])
```

I don't really have much time to make a PR with tests and all that right now, but hopefully it's helpful.

**Comment by arthur-e** [comment]
FYI: Until this fix is available (ipython 7.16.1 is the latest available to me right now), end-users can avoid these deprecation warnings by forcing an install of jedi version 0.15.2.

**Comment by Carreau** [comment]
> FYI: Until this fix is available (ipython 7.16.1 is the latest available to me right now), end-users can avoid these deprecation warnings by forcing an install of jedi version 0.15.2.

Or just set the DeprecationWarnings to be ignored...

https://docs.python.org/3/library/warnings.html#overriding-the-default-filter

Also 7.17 is out since 8 days ago.
